webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx

```diff
@@ -7,25 +7,31 @@ import Menu from 'components/widgets/menu/menu';
 
 import {localizeMessage} from 'utils/utils';
 
-type Props = {
+interface CloseChannelProps {
     isArchived: boolean;
     actions: {
         goToLastViewedChannel: () => void;
     };
 }
 
-export default class CloseChannel extends React.PureComponent<Props> {
-    private handleClose = () => {
-        this.props.actions.goToLastViewedChannel();
+const CloseChannel = ({
+    isArchived,
+    actions,
+}: CloseChannelProps): JSX.Element => {
+    const handleClose = () => {
```

Comment:
This is creating a new function on every render, potentially re-rendering the underlying component. We could use `useCallback` here, but the callback is so simple, we can use it directly as a prop for the next component. i.e.

```
<Menu.ItemAction
    onClick={actions.goToLastViewedChannel}
/>
```